### DISCUSSION TITLE
Changelog language variants

### DISCUSSION THREAD

**Discussion by MichalBryxi** [olivierlacan/keep-a-changelog#496] [Ideas]
As a project developer, I want to provide changelog to other developers (by convention in English) and to clients/end consumers (in their own native language). So that both developers and clients can consume the changelog without any hassle.

Just noticed that there is no recommendation when it comes to such situation.
 * Should I write everything in one file and just structure it? How?
 * Should I write each language to it's own file? What should be the structure?
   * `CHANGELOG.en.md`, `CHANGELOG.cs.md`
   * `CHANGELOG.md` (english by default - good for backwards compatibility), `CHANGELOG.cs.md`
   * ???
 * What if I write only one changelog - in my own language? Would it not be more convenient to also mark the file as such? So that everybody can see at a glance that it is not in English?

There are probably more questions to ask. Just wanted to start the conversation. Anybody has opinions?

**Comment by polarathene** [comment]
I'd go with `CHANGELOG.md` for english, then a separate changelog with language extension in the file like you suggested.

**Comment by mitelg** [comment]
I would also like to know how to do that in a "good" way. Creating a new file for each language has one problem for me: you have to remember to maintain 1..n files for each change. I'm open for suggestions :blush:

**Comment by polarathene** [comment]
> Creating a new file for each language has one problem for me: you have to remember to maintain 1..n files for each change

vs a single file and remembering to update each language? With multiple files you'd get visual confirmation with git when that language was last updated or if you have n languages that the same amount in files are going to be committed. If it helps organize, place them all in changelog directory for additional clarity that the amount of changelog/language files being updated is the same total in that directory.

Not sure how you get similar benefits from a single file. As a user maintaining or viewingthe language of a changelog, should you want to contribute an improvement/fix or go over that languages history(assuming you have a community providing the translations), then separate files also allow for keeping the history of a specific language clean. The history isn't mixed with other languages. Provided the changelog commits for each file all have their updates between release tags, you get clean history across them all even if they're not being all updated/committed at once.

**Comment by mitelg** [comment]
@polarathene thank you very much for your reply :+1: I think you convinced me :blush:
